runtime/common/src/origins.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+/// Custom origins for governance interventions.
+pub mod gov {
+	pub use pallet_custom_origins::*;
+
+	#[frame_support::pallet]
+	pub mod pallet_custom_origins {
+		use frame_support::pallet_prelude::*;
+
+		#[pallet::config]
+		pub trait Config: frame_system::Config {}
+
+		#[pallet::pallet]
+		pub struct Pallet<T>(_);
+
+		#[derive(PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo, RuntimeDebug)]
+		#[pallet::origin]
+		pub enum Origin {
```

Comment:
To understand this pallet correctly. The idea behind this is to add these variants to the composed `RuntimeOrigin` automatically once this pallet is added by the runtime, right?

Pretty smart!